lib/rexml/validation/relaxng.rb

```diff
@@ -262,9 +262,10 @@ def next( event )
           rv = super
           return rv if rv
           @prior = @previous.pop
-          return @prior.next( event )
+          @prior.next( event )
+        else
+          super
         end
-        super
```

Comment:
Former construction was odd, since it could be mistakenly read as calling `super` twice. Keeping `super` in the else statement prevents this misreading.